go/vt/vtgate/vcursor_impl.go

```diff
@@ -556,9 +556,18 @@ func (vc *vcursorImpl) ExecuteKeyspaceID(keyspace string, ksid []byte, query str
 		Sql:           query,
 		BindVariables: bindVars,
 	}}
+	if !rollbackOnError {
+		vc.safeSession.queryFromVindex = true
+		defer func() {
+			vc.safeSession.queryFromVindex = false
+		}()
+	}
```

Comment:
I don't understand this change. Maybe we should add a comment that explains why we do this